charms/tensorboard-controller/metadata.yaml

```diff
@@ -1,14 +1,19 @@
-# Copyright 2021 Canonical Ltd.
+# Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 name: tensorboard-controller
 description: Kubeflow Tensorboard Controller
 summary: Kubeflow Tensorboard Controller
-min-juju-version: '2.9.2'
-series: [kubernetes]
+website: https://charmhub.io/tensorboard-controller
+source: https://github.com/canonical/kubeflow-tensorboards-operator/
+issues: https://github.com/canonical/kubeflow-tensorboards-operator/issues
+docs: https://discourse.charmhub.io/t/charmed-tensorboard-controller/8239
+containers:
+  tensorboard-controller:
+    resource: tensorboard-controller-image
 resources:
-  oci-image:
+  tensorboard-controller-image:
```

Comment:
I think this change is the cause of your [publish CI failing](https://github.com/canonical/kubeflow-tensorboards-operator/actions/runs/5111763643/jobs/9189111793?pr=69#step:5:71).  The upload-charm action has an [annoying problem where it cannot upload if a resource does not already exist](https://github.com/canonical/charming-actions/issues/76).  Basically it just means you need to manually publish the resource once, then the CI will work

This change *might* have implications for upgrading your charm, too, but I don't think it does.  Just something to keep an eye on when you test upgrades

Reply:
This change also breaks the joint integration tests [here](https://github.com/canonical/kubeflow-tensorboards-operator/actions/runs/5111763643/jobs/9189086326?pr=69#step:4:75), so we need to fix that up too.  Should be easy enough.

Will also need to add `--trust` to the deployment there I expect, and any other changes you see needed

Reply:
Both of the above make total sense! I hadn't noticed these tests existed until I saw them failing. I think it's worth having a more descriptive name there (`oci-image` feels like a reserved keyword, something that the framework needs there), with the same rationale applying to the container name. I'll fix any issues with the tests (and test the upgrade)!

Reply:
yeah I think the `oci-image` name came from a time when that was treated like a keyword by some additional tooling (they'd always look for that key as **the** resource and do stuff with it), but I dont think we have anything like that anymore